File: memory-domain/src/main/java/com/memory/document/memory/MemoryDocumentRepository.java
package com.memory.document.memory;

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MemoryDocumentRepository extends ElasticsearchRepository<MemoryDocument, String>, MemoryDocumentRepositoryCustom {
    
    void deleteByMemoryId(Long memoryId);
    
    MemoryDocument findByMemoryId(Long memoryId);
}

File: memory-domain/src/main/kotlin/com/memory/document/memory/MemoryDocumentRepository.kt
package com.memory.document.memory

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
import org.springframework.stereotype.Repository

@Repository
interface MemoryDocumentRepository : ElasticsearchRepository<MemoryDocument, String>, MemoryDocumentRepositoryCustom {

    fun deleteByMemoryId(memoryId: Long)

    fun findByMemoryId(memoryId: Long): MemoryDocument
}
